src/SignupWithMobile.js

```diff
@@ -288,12 +292,17 @@ export default class SignupWithMobile extends React.PureComponent<
             />
           </View>
           <TouchableOpacity
+            disabled={loading}
             style={[defaultStyles.submitButton, submitButtonStyle]}
             onPress={this.onPressSubmitButton}
           >
-            <Text style={[defaultStyles.submitText, submitButtonTextStyle]}>
-              {submitButtonText}
-            </Text>
+            {loading ? (
+              <ActivityIndicator />
+            ) : (
+              <Text style={[defaultStyles.submitText, submitButtonTextStyle]}>
+                {submitButtonText}
+              </Text>
+            )}
```

Comment:
According to @frank-oursky , we do not need to show `ActivityIndicator` to indicate loading state. Just make loading state same as disabled state. That is, when it is either loading or disabled, the button should be semi-transparent.

In this way, we can avoid layout instability and it is very easier to implement.